src/targets/gpu/include/migraphx/gpu/hip.hpp

```diff
@@ -63,7 +63,7 @@ struct hip_sync
         return pack(f(self.tag, "tag"));
     }
 
-    std::string name() const { return "hip::sync"; }
+    std::string name() const { return "hip::sync_device"; }
     shape compute_shape(const std::vector<shape>& inputs) const
     {
         if(inputs.empty())
```

Comment:
So this should always return `{}` so dead code elimination wont remove it regardless of inputs.

Reply:
Fixed, and add a sync_device pass after the schedule pass to avoid a bunch of other issues.